pagopa/io-app
# chore(IT Wallet): [SIW-1868] Add full screen skeumorphic card modal

## Short description
This PR introduces a feature that allows users to view a credential's skeuomorphic card in full-screen mode. This mode can be activated by tapping the card on the credential details screen.

## List of changes proposed in this pull request
- Added `useScaleAnimation` hook (to be removed once this PR is merged: https://github.com/pagopa/io-app-design-system/pull/358)
- Added `FlipGestureDetector` component, which handles swipe gestures to flip the skeumorphic cards in the credential details screen
- Updated `FlippableCard` component in order to handle an optional `orientation` prop, which changes the axis type for the flip animation
- Updated `ItwSkeumorphicCard` component in roder to handle an optional `orientation` prop, which rotates the card and adjust its scale to fit the screen correctly.
- Refactored `getCredentialStatus` and `itwCredentialStatusSelector`:
  - Extracted logic from  `itwCredentialStatusSelector` and moved it to `getCredentialStatusObject`, `getCredentialStatusObject` can be freely used without relying to the redux store and `itwCredentialStatusSelector` for the memoized value
  - Moved `getCredentialStatus` and `getCredentialStatusObject` to `itwCredentialStatusUtils.ts` file
  - Added `ItwPresentationCredentialCardModal` in the IT Wallet navigation stack
  - Updated IT Wallet playgrounds to test the new feature

## How to test
1. Navigate to the credential details screen or the IT Wallet playground in the settings screen.
2. Tap on a skeuomorphic credential card to view it in full-screen mode.
3. Verify the following:
    - The card is rendered correctly in full-screen mode.
    - The screen brightness gently increases when entering full-screen mode.
    - The screen brightness reverts to its original value upon exiting full-screen mode.

## Preview

https://github.com/user-attachments/assets/66743a6f-ea96-4de3-a150-f88ccfa30eea
